### PR TITLE
Admin merchant update

### LINKED ISSUE #15: Admin Merchant Update
As an admin,
When I visit a merchant's admin show page
Then I see a link to update the merchant's information.
When I click the link
Then I am taken to a page to edit this merchant
And I see a form filled in with the existing merchant attribute information
When I update the information in the form and I click ‘submit’
Then I am redirected back to the merchant's admin show page where I see the updated information
And I see a flash message stating that the information has been successfully updated.

### PR DESCRIPTION
**Pull request by ParkerLockhart**
PR includes: 

Fixes for date formatting in controller/tests
Fixes for mistake in test variable names

Tests, routing, controller methods, and views for admin merchant update functionality

Closes #15

### CODE REVIEW

**Review by joel-grant — APPROVED** [review]
I like the form_with model you set up!

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -5,5 +5,25 @@ def index
...
+  end
+
+  def update
+    merchant = Merchant.find(params[:id])
+    if merchant.update(merchant_params)
+      redirect_to "/admin/merchants/#{merchant.id}"
+      flash[:alert] = "Successfully Updated Item"
+    else
+      redirect_to "/admin/merchants/#{merchant.id}/edit"
+      flash[:alert] = "Error: #{error_message(merchant.errors)}"
```
Comment (kbhoffmann) [inline app/controllers/admin/merchants_controller.rb:21]: Great use of error messages for the flash message.

File: app/models/invoice.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Invoice < ApplicationRecord
   }
 
   def pretty_created_at
-    created_at.strftime("%A, %B%-d, %Y")
+    created_at.strftime("%A, %B %-d, %Y")
```
Comment (kbhoffmann) [inline app/models/invoice.rb:16]: I didn't catch this on the last pull request review.  I'm sure its something you noticed when you looked at the page, which I didn't actually do!

File: spec/features/merchants/items/index_spec.rb [hunk 3]
```diff
@@ -14,8 +14,8 @@
   end
 
   it 'the item name is a link to the merchant items show page' do
-    click_link("#{@item1.name}")
+    click_link("#{@item.name}")
```
Comment (kbhoffmann) [inline spec/features/merchants/items/index_spec.rb]: This looks like it fixes an issue I had come across this morning. Thank you for fixing it!

**Review by kbhoffmann — APPROVED** [review]

**Comment by kbhoffmann** [comment]
I just saw the merge conflict Parker.  Is that something you are able to resolve?

File: spec/features/merchants/items/index_spec.rb (hunk 3 above)
Comment (ParkerLockhart) [reply spec/features/merchants/items/index_spec.rb]: It popped up when I ran rspec and I thought it was a mistake I had made earlier, then when I thought about it later I wasn't sure if I accidentally changed something one of y'all wrote  😅 
Either way I'm glad its good now!

**Comment by ParkerLockhart** [comment]
> 

Should be good to go now!

**Pull request merged**